eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -0,0 +1,48 @@
+param(
+  $RepoOwner,
+  $RepoName,
+  $BranchPrefix,
+  $AuthToken
+)
+
+. "${PSScriptRoot}\common.ps1"
+
+LogDebug "Operating on Repo [ $RepoName ]"
+try{
+  $branches = (List-References -RepoOwner $RepoOwner -RepoName $RepoName -Ref "heads/$BranchPrefix").ref
+}
+catch {
+  LogError "List-References failed with exception:`n$_"
+  exit 1
+}
+
+foreach ($branch in $branches)
+{
+  try {
+    $branchName = $branch.Replace("refs/heads/","")
+    $head = "${RepoOwner}/${RepoName}:${branchName}"
+    LogDebug "Operating on branch [ $branchName ]"
+    $pullRequests = List-PullRequests -RepoOwner $RepoOwner -RepoName $RepoName -head $head
+  }
+  catch
+  {
+    LogError "List-PullRequests failed with exception:`n$_"
+    exit 1
+  }
+
+  "bvranch $branch"
+  "PR COunt $($pullRequests.Count)"
+
+
+  if ($pullRequests.Count -eq 0)
+  {
+    LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated Pull Request. Deleting Branch"
+    try{
+      Delete-References -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
```

Comment:
In addition to deleting the reference, should we also delete the pull request itself or will that happen automatically?

Reply:
Am not sure what you mean? Do we want to delete the pull request. Isn't it better to keep that after merging it.

Reply:
Sorry, I was thinking that this was a a clean-up just incase we wanted to abort. I take back my feedback ;)